### ISSUE TITLE
lastSyncTime is not being used (sometimes)

### ISSUE THREAD

**Issue by airstance** [awslabs/aws-mobile-appsync-sdk-ios#232]
**Describe the bug**
There are times when the delta sync code does not use the correct lastSyncTime

**To Reproduce**
1. Start a sync operation with a delta sync query
2. Perform at least one sync from the server so that lastSyncTime is set
3. Force a new sync operation (rerun application, foreground switch, network toggle)
4. There are times when the lastSyncTime isn't fetched from the cache correctly.  It will look somewhat random.

**Expected behavior**
The lastSyncTime should be the same value as the last successful delta/base sync

**Environment(please complete the following information):**
 - AppSync SDK Version: 2.10.3
 - Dependency Manager: Cocoapods
 - Swift Version : 4.2+

**Additional context**
There's a bug in `AppSyncSubscriptionWithSync.getOperationHash()`.  The code uses the description of dictionaries which have no order and can randomly return different string values.

Here's a fix that first sorts the dictionary into an array.

```
    private func getOperationHash() -> String {
        
        var baseString = ""
        
        let variables = baseQuery.variables?.sorted(by: { $0.0 < $1.0 }).description ?? ""
        baseString = type(of: baseQuery).requestString + variables

        if let subscription = subscription {
            let variables = subscription.variables?.sorted(by: { $0.0 < $1.0 }).description ?? ""
            baseString = type(of: subscription).requestString + variables
        }
        
        if let deltaQuery = deltaQuery {
            let variables = deltaQuery.variables?.sorted(by: { $0.0 < $1.0 }).description ?? ""
            baseString = type(of: deltaQuery).requestString + variables
        }
        AppSyncLog.debug("Creating hash for \(baseString)")
        return AWSSignatureSignerUtility.hash(baseString.data(using: .utf8)!)!.base64EncodedString()
    }
```

This was the issue that was causing me trouble with #219 

These bugs are very time consuming to find and fix.

**Comment by rohandubal** [comment]
Thanks for the detailed information @airstance 

I am working to reproduce this on my side.

**Comment by royjit** [comment]
This fix has been released in the version 2.14.0
